libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java

```diff
@@ -245,6 +245,7 @@ boolean isMuted() {
   void stopNavigation() {
     navigation.unregisterRouteProgressObserver(navigationViewVm);
     navigation.unregisterLocationObserver(navigationViewVm);
+    navigation.unregisterOffRouteObserver(offRouteObserver);
```

Comment:
since we register the observer when startNavigation, we should unregister it when stop.